### ISSUE TITLE
Indexer is unable to see files with the character "【" in them

### ISSUE THREAD

**Issue by victorhooi** [stashapp/stash#572]
**Describe the bug**
The Stash scan seems to ignore folders with the character "【" in them.

Is this related to Unicode maybe, or how we scan?

**To Reproduce**
1. Add a video/scene with "【" in the filename.
2. Run a scan operation from the Stash Web UI,
3. Attempt to search for the video by name.

**Expected behavior**
The Stash scan operation should pickup all subdirectories in the specified download directory.

**Screenshots**
If applicable, add screenshots to help explain your problem please ensure that your screenshots are SFW or at least appropriately censored.

**Stash Version: (from Settings -> About):**
```
Version:	v0.1.1-148-gd74a303
Build hash:	d74a303
Build time:	2020-05-14 00:55:16
```

**Desktop (please complete the following information):**
 - OS: Debian Buster
 - Browser: Chrome
 - Version: 85.0.4151.0

**Comment by victorhooi** [comment]
It appears to be this character

https://unicodelookup.com/#%E3%80%90/1

which is a fairly common punctuation character in Asian languages:

https://en.wikipedia.org/wiki/Japanese_punctuation
https://en.wikipedia.org/wiki/Chinese_punctuation

Also - to clarify, that character appears in both the parent directory of the video, and also in the video filename itself.

**Comment by victorhooi** [comment]
Upon further testing, it only happens if the bracket character is in the video filename as well.

An example of a full non-working filename is this:

```
【ses23.com】BOMN-122.1080p/【ses23.com】BOMN-122.1080p.mkv
```

To workaround the issue - you can move the file (e.g. introduce a space in front of the bracket) - however, this will break things that depend on the original filename (e.g. seeding):
```
cp -R '【ses23.com】BOMN-122.1080p/' ' 【ses23.com】BOMN-122.1080p/'
cd ' 【ses23.com】BOMN-122.1080p'/
mv '【ses23.com】BOMN-122.1080p.mkv' ' 【ses23.com】BOMN-122.1080p.mkv'
```

**Comment by victorhooi** [comment]
As a temporary workaround - you can make a symlink to the video file, to either introduce a space before "【", or rename the file otherwise:

```
pwd
/home/name/qbittorrent/downloads/【ses23.com】example.1080p
ln -s 【ses23.com】example.1080p.mkv " 【ses23.com】example.1080p.mkv"
```

bnkai mentioned that the third-party globbing library we use is this one:

https://github.com/bmatcuk/doublestar

**Comment by bnkai** [comment]
it seems to be related to the doublestar library
the closest to how we use it is the code below
```
package main

import (
        "fmt"
        "github.com/bmatcuk/doublestar"
        "path/filepath"
)

func main() {
        path := "/home/username/test"
        pattern := "**/*.{mp4}"
        pattern = filepath.Join(path, pattern)
        re, _ := doublestar.Glob(pattern)

        for _, file := range re {
                fmt.Println(file)
        }
}
```

now if you do 
```
touch "/home/username/test/file.mp4"
touch "/home/username/test/【file.mp4"
touch "/home/username/test/ 【file.mp4"
```
and run the code only the first and last file get matched and printed

**Comment by InfiniteTF** [comment]
Yeah it seems to be a doublestar bug. When iterating through the filename the library uses the width of the first character it encounters as a step, so if it's whitespace or an ascii character the step is 1, if it's wide like with the failing test case, the step is 2. This basically only works for pure ascii or pure utf16/32 characters, not utf8 which can have mixed widths.

**Comment by bnkai** [comment]
Yes i made an issue reporting the case. I guess we have to wait for that.

**Comment by victorhooi** [comment]
@bnkai Do you happen to know if https://github.com/mattn/go-zglob has the same issue as the doublestar library?

**Comment by bnkai** [comment]
@victorhooi it doesn't seem to support a feature we use https://github.com/mattn/go-zglob/issues/14 
and i don't want to have to verify that they give the same results.
If the other issue takes long to be resolved though i'll do that since it  seems to work correctly for the "【" case ( it helps that the dev is japanese :-) ) .

**Comment by bnkai** [comment]
@victorhooi the doublestar library just got a fix. I'll give it a try and make a pr if it works as intended.

**Comment by WithoutPants** [comment]
Fixed via #586
